### PR TITLE
Add API to allow better tracking consumers

### PR DESCRIPTION
**Pull request by thiblahute**
See commit messages.

### CODE REVIEW

File: plugins/src/webrtcsink/imp.rs [hunk 1]
```diff
@@ -1716,7 +1719,7 @@ impl WebRTCSink {
             }
         })?;
 
-        element.emit_by_name::<()>("new-webrtcbin", &[&peer_id, &webrtcbin]);
+        element.emit_by_name::<()>("consumer-added", &[&peer_id, &webrtcbin]);
```
Comment (MathieuDuponchelle) [inline plugins/src/webrtcsink/imp.rs:1722]: so, I see you refactored the consumer removal code path to emit the signal without the lock, should we move that code to the end of the function and add a `drop (state)` before?

File: plugins/src/webrtcsink/imp.rs [hunk 2]
```diff
@@ -2457,22 +2463,60 @@ impl ObjectImpl for WebRTCSink {
...
+                    glib::types::Type::UNIT.into(),
+                )
+                .build(),
+
+                /*
+                 * RsWebRTCSink::consumer_removed:
+                 * @consumer_id: Identifier of the consumer that was removed
+                 * @webrtcbin: The webrtcbin connected to the newly removed consumer
+                 *
+                 * This signal is emitted right after the connection from a consumer
```
Comment (MathieuDuponchelle) [inline plugins/src/webrtcsink/imp.rs:2488]: connection with?

**Comment by MathieuDuponchelle** [comment]
lgtm otherwise, tnx :)

File: plugins/src/webrtcsink/imp.rs (hunk 1 above)
Comment (thiblahute) [reply plugins/src/webrtcsink/imp.rs:1722]: Good catch! Fixed.

**Comment by thiblahute** [comment]
Also passed the code to `cargo fmt` as I introduced some "regressions" implementing the navigation interface.

File: plugins/src/webrtcsink/imp.rs [hunk 3]
```diff
@@ -362,7 +360,8 @@ fn setup_encoding(
             // to incorrect color display in Chrome (but interestingly not in
             // Firefox). In any case, restrict to exclude RGB formats altogether,
             // and let videoconvert do the conversion properly if needed.
-            structure_builder = structure_builder.field("format", &gst::List::new(&[&"NV12", &"YV12", &"I420"]));
+            structure_builder =
+                structure_builder.field("format", &gst::List::new(&[&"NV12", &"YV12", &"I420"]));
```
Comment (sdroege) [inline plugins/src/webrtcsink/imp.rs:364]: Why all the `&` :sob:
They'are all not needed and just add line-noise

File: plugins/src/webrtcsink/imp.rs [hunk 4]
```diff
@@ -2457,22 +2469,67 @@ impl ObjectImpl for WebRTCSink {
         static SIGNALS: Lazy<Vec<glib::subclass::Signal>> = Lazy::new(|| {
             vec![
                 /*
-                 * RsWebRTCSink::new-webrtcbin:
-                 * @peer_id: Identifier of the peer associated with the consumer added
+                 * RsWebRTCSink::consumer-added:
```
[the remark this answers is not in the document]
Comment (MathieuDuponchelle) [reply plugins/src/webrtcsink/imp.rs:2472]: Not that I know of, wishful placeholders as far as I understand :)

File: plugins/src/webrtcsink/imp.rs (hunk 3 above)
Comment (MathieuDuponchelle) [reply plugins/src/webrtcsink/imp.rs:364]: not @thiblahute's fault, probably just me writing this code as it once needed to be written?

Comment (sdroege) [reply plugins/src/webrtcsink/imp.rs:364]: Yeah, that was needed until 2-3 months ago

File: plugins/src/webrtcsink/imp.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (thiblahute) [reply plugins/src/webrtcsink/imp.rs:2472]: Getting hotdoc to generate doc for rust elements should be pretty straight forward.

[the remark this answers is not in the document]
Comment (MathieuDuponchelle) [reply plugins/src/webrtcsink/imp.rs:2472]: depends on whether a new comment parser needs to be written, but yes if we're satisfied with gtk-doc syntax it should be a simple matter of running the scanner and build system plumbing basically

**Pull request merged**